### ISSUE TITLE
Remove bin/plugin update command without arguments

### ISSUE THREAD

**Issue by suyograo** [elastic/logstash#4415]
`bin/plugin update` when run on LS instance will update all the default plugins bundled to the latest versions published on RubyGems.org. While this may seem useful (to uptake latest bug fixes), plugin development and release is isolated w.r.t. LS core. Mass update of all default plugins has the potential to cause dependency issues. This is because LS release package pins the version numbers of dependency in Gem lock file and update does not seem to properly handle updates to dependency. See discussion here: https://discuss.elastic.co/t/logstash-2-1-fails-to-start-after-logstash-output-elasticsearch-update/38330/2 and https://github.com/logstash-plugins/logstash-output-elasticsearch/issues/335
which was caused by 
https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/334

I was able to reproduce this behavior with LS 2.1.1 and running `bin/plugin update`. It updated logstash-output-elasticsearch to 2.3.1 among other plugins.

When I manually update the single plugin:
```
bin/plugin update logstash-output-elasticsearch
```

it updates it to 2.3.0 which is a good version (without dependency conflict)

I am recommending we discourage and remove the use of `bin/plugin update` and only allow specific updates to plugins. Also, given the recent LS release cadence, users do not have to run an update to get the latest big fixes to plugins.

Thoughts?

**Comment by andrewvc** [comment]
Am I correct in my understanding that `bin/plugin update` is different than `bundle update`? IIRC `bundle update` gets the lowest common denominator of all libs. Does our code do something different?

I do think this is a useful task to have for our own internal purposes, but then we could just run `bundle update` ourselves yes?

**Comment by aschaef** [comment]
I'd be happy upgrading individual plugins. I ran into the issue with the logstash-output-elasticsearch 2.3.1 upgrade being new to the product and only finding documentation referring to the 'bin/plugin update' command. Having the ability to upgrade individual plugins would give us the granularity we'd need going forward.

**Comment by jordansissel** [comment]
@suyograo I think a general case of 'bin/plugin update' that works correctly would be great, but as you describe, currently this doesn't work well enough. Given that, I am in favor of making 'bin/plugin update' with no arguments cause a usage error ("please specify plugins to update") until some future time where we make it possible for 'bin/plugin update' to do the right thing for a blanket upgrade.

+1 on removing blanket upgrade for now.

**Comment by suyograo** [comment]
@jordansissel correct, I am advocating removing global upgrade until we have it bullet-proof. It would be disastrous for users to run this on a production system and there is no roll-back from there. Only alternative if something is hosed is to reinstall binaries from scratch

**Comment by purbon** [comment]
> Am I correct in my understanding that bin/plugin update is different than bundle update? IIRC bundle update gets the lowest common denominator of all libs. Does our code do something different?

This days ```bin/plugin update``` does the sames as ```bundler update``` call, the only difference is that we remove any version constraints for plugins, so update the last version available always (see https://github.com/elastic/logstash/blob/master/lib/pluginmanager/update.rb#L42-L44) for more details.

@andrewvc 

@suyograo @jordansissel I agree with you two, enforcing people to use single plugin update makes a lot of sense to me. I would also advocate for reviewing/revisiting the way we set dependencies constraints for our plugins, might be we can help also the update process there.
